src/main.ts

```diff
@@ -26,12 +27,41 @@ async function run(): Promise<void> {
   try {
     // Get action inputs.
     let projectId = core.getInput('project_id');
+    const cwd = core.getInput('working_directory');
     const deliverables = core.getInput('deliverables');
     const imageUrl = core.getInput('image_url');
     const version = core.getInput('version');
     const promote = core.getInput('promote');
     const serviceAccountKey = core.getInput('credentials');
 
+    // Change working directory
+    if (cwd) {
+      try {
+        process.chdir(cwd.trim());
+      } catch (error) {
+        core.setFailed(error.message);
```

Comment:
Shouldn't we just throw an error with the correct message and let the parent try/catch set to failed?

Reply:
Thanks for catching